Firebase/Core/FIRApp.m

```diff
@@ -227,6 +232,7 @@ - (void)deleteApp:(FIRAppVoidBoolCallback)completion {
     if (sAllApps && sAllApps[self.name]) {
```

Comment:
self.name can be nil. We should check it before accessing the dictionary.

Reply:
We throw an exception if a user tries to configure the app with a `nil` or empty name, which should catch this. If you're still concerned, we can add it in a separate PR since it's unrelated to this change.

Reply:
ok. I think we should be careful of accessing dictionary when the provided key is nil. I'm concerned that self.name might be nil at some point during the entire app life cycle.